README.md

```diff
@@ -53,7 +53,47 @@ cd onyo
 pip install -e ".[tests, docs]"
 ```
 
-### Tests
+### Design
+
+Onyo follows a set of development and coding conventions throughout its code
+base. For linting and type checking, the Onyo project uses flake8 and Pyre. The
+following sections describe additional conventions followed in this project,
+extending the existing [PEP 8 Style Guide](https://peps.python.org/pep-0008/).
+
+#### The Complete Code Base:
+- All classes, functions and properties have a doc-string describing their
+  functionality.
+- All functions have type hinting (including explicit return type for `None`).
+- The code conforms to [PEP 8](https://peps.python.org/pep-0008/).
+
+#### lib Tests (`tests/lib/`):
+- follow Onyo's code conventions, see [The Complete Code Base](#the-complete-code-base).
+- define the expected behavior of the code for all use-cases of the different
```

Comment:
nit: "of the code" is a generic statement. It's ok to omit it, and the sentence is clearer as a result:

> define the expected behavior for all use-cases of the different consumers

However, I also struggle with the word "define" here. The /definition/ should be in the function's docstring. The tests should make sure that definition is well exercised.

Reply:
While "define" might not be the perfect term, I feel like "make sure that definition from the function's doc-strings is well exercised" just does not sound that clear.

And I want to mention that the tests are implemented as independently as possible from the code, which is were the docstring is. I know that this might be different for the public functions of the repo etc, but still I feel like "define" is at least ok.